docs/cpan-starjacking.md

```diff
@@ -0,0 +1,120 @@
+
+## How to choose what CPAN module to depend on?
+CPAN has such a huge collection of modules available it's sometimes difficult to make a decision about the best one to use.
+
+Some lists are available like [Task::Kensho](https://metacpan.org/pod/Task::Kensho), [Awesome Perl](https://github.com/hachiojipm/awesome-perl) or [Awesome Perl](https://github.com/uhub/awesome-perl) (again) but they definitely don't cover all needs.
+
+Naming of modules can also help, for instance `Mojo::*` modules should be, by convention, mainly using other `Mojo` inner circle modules as dependencies. `*::Tiny` are generally smaller alternative to other modules of the same purpose, etc...
+
+Most of the time and when possible, users will choose modules and authors that have an established "reputation". This reputation gives them authority on a topic, and the trust of users in term of security.
+
+On top of that, some indicators of quality (build reports and known problems) and activity (maintainance) can be taken into account.
+
+## Popularity indicators with CPAN
+Beyond reputation (module and/or author), MetaCPAN provides some popularity indicators for CPAN modules:
+- ++
+![plusplus](../media/starjacking/plusplus.png)
+
+- [river](https://neilb.org/2015/04/20/river-of-cpan.html) position
+![river](../media/starjacking/river.png)
+
+## Quality indicators in CPAN
+Some other indicators can help in the choice but give more informations about *quality*:
+- Issues
+![Issues](../media/starjacking/issues.png)
+- CPANTesters
+![CPAN Testers](../media/starjacking/testers.png)
+- Bus factor
+![Bus factor](../media/starjacking/bus.png)
+
+## StarJacking ![Stars](../media/starjacking/stars.png)
+StarJacking refers to the trust that VCS stars provide to a module.
+
+By reusing the popularity of another project, unrelated to the module,
+and exploiting the non-verification of the link (metadata),
+an attacker can make a module appear much more popular than it actually is.
+
+Other ecosystems like PyPI and npm are also impacted by this problem.
+
+Like many other programming language ecosystems, we can attach "metadata"
+such as a public repository, website and issue tracker URLs to CPAN packages.
+
+It has to be done directly in the release metadata.
+
+Example of using resources metadata pointing to a popular project, for instance in `META.yml`
+```
+META_MERGE   => {
+    resources => {
+        repository => {
+            type => 'git',
+            url  => 'https://github.com/popular/lot-of-stars.git',
+            web  => 'https://github.com/popular/lot-of-stars',
+        },
+        bugtracker => {
+            web => 'https://github.com/popular/lot-of-stars-and-tickets',
+        },
+    },
+},
+```
+
+Read [CPAN add link to version control system](https://perlmaven.com/how-to-add-link-to-version-control-system-of-a-cpan-distributions) for more about how to set the metadata.
```

Comment:
Could we prefer to link to primary sources first, and then add links like this one under "further reading" as supporting material? I think that would help make this study come across as more professional.

Reply:
Used official documentation of CPAN::META::Spec instead